helix-core/src/main/java/org/apache/helix/controller/rebalancer/util/WagedRebalanceUtil.java

```diff
@@ -61,7 +62,11 @@ public static Map<String, Integer> fetchCapacityUsage(String partitionName,
       ResourceConfig resourceConfig, ClusterConfig clusterConfig) {
     Map<String, Map<String, Integer>> capacityMap;
     try {
-      capacityMap = resourceConfig.getPartitionCapacityMap();
+      if (resourceConfig != null) {
+        capacityMap = resourceConfig.getPartitionCapacityMap();
+      } else {
+        capacityMap = new HashMap<>();
+        }
```

Comment:
NIT: can be simplified as to one-liner

capacityMap = resourceConfig == null ? new HashMap<>() : resourceConfig.getPartitionCapacityMap();